### PR TITLE
Added states to File.

### PR DESCRIPTION
**Pull request by mickel1138**
Added status to File. Also migration changes.

Let me know what you guys things of the Status naming?

Don't forget to run/test a migration or do load dump/

### CODE REVIEW

File: src/Entity/File.php [hunk 1]
```diff
@@ -11,6 +11,26 @@
...
+
+    /**
+     * Status value for a File that has been deleted. (File still exists on disk!)
+     */
+    const FILE_DELETED = 'deleted';
+
+    /**
+     * Status value for a File that has been proccessed.
+     */
+    const FILE_DONE = 'done';
```
Comment (praneethpr) [inline src/Entity/File.php:27]: I think processed would be a better name for it, because all other names are action names

**Review by praneethpr — APPROVED** [review]

File: src/Migrations/Version20200625131822.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ public function up(Schema $schema) : void
...
             $this->addSql("UPDATE dataset_submission SET fileset_id = currval('public.fileset_id_seq') WHERE id = $id");
-            $this->addSql("INSERT INTO file (id, fileset_id, creation_time_stamp, creator_id, modification_time_stamp, modifier_id, file_name, file_size, file_sha256_hash, file_path) SELECT nextval('file_id_seq'), currval('public.fileset_id_seq'), CURRENT_TIMESTAMP(0), 0, CURRENT_TIMESTAMP(0), 0, dataset_file_name, dataset_file_size, dataset_file_sha256_hash, SUBSTRING(dataset_file_uri FROM '\/[^\/].*$') FROM dataset_submission WHERE id = $id");
+            $this->addSql("INSERT INTO file (id, fileset_id, creation_time_stamp, creator_id, modification_time_stamp, modifier_id, file_name, file_size, file_sha256_hash, file_path, status) SELECT nextval('file_id_seq'), currval('public.fileset_id_seq'), CURRENT_TIMESTAMP(0), 0, CURRENT_TIMESTAMP(0), 0, dataset_file_name, dataset_file_size, dataset_file_sha256_hash, SUBSTRING(dataset_file_uri FROM '\/[^\/].*$'), 'processed' FROM dataset_submission WHERE id = $id");
```
Comment (praneethpr) [inline src/Migrations/Version20200625131822.php:30]: should be renamed to "done"

**Review by fightingtexasaggie — COMMENTED** [review]
fos:elastica:populate currently fails with "Undefined class constant 'self::FILE_UPLOADED'"

**Review by fightingtexasaggie — APPROVED** [review]
ok, working now

**Review by praneethpr — APPROVED** [review]

**Pull request merged**